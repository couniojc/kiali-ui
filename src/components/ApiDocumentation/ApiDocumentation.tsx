// Only REST is supported for now, we're working on GraphQL !
//
// For REST, we use the excellent https://github.com/mrin9/RapiDoc
// https://github.com/mrin9/RapiDoc/blob/master/LICENSE.txt
//
// For now the 'allow-try' is set to false, but a PR will follow
// with an optional proxy to run requests against the service

import * as React from 'react';
import { config } from '../../config';
import 'rapidoc/dist/rapidoc-min.js';

/* eslint-disable */
declare global {
  namespace JSX {
    interface IntrinsicElements {
      'rapi-doc': any;
    }
  }
}
/* eslint-enable */

interface ApiDocumentationProps {
    apiType: string;
    namespace: string;
    service: string;
  };

const urls = config.api.urls;

export class ApiDocumentation extends React.Component<ApiDocumentationProps> {

  render() {
    return (
      <div>
        <rapi-doc
           spec-url={urls.serviceApiDocumentation(this.props.namespace,this.props.service)}
           show-header="false"
           show-info="false"
           allow-authentication="false"
           allow-try="false"
           layout="column"
        />
     </div>
    );
  }
}
