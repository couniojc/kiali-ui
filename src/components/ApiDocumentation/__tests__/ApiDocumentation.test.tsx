import * as React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { ApiDocumentation } from '../ApiDocumentation';

describe('ApiDocumentation', () => {
  it('renders when not known', () => {
    let wrapper = shallow(<ApiDocumentation apiType="rest" namespace="ns1" service="serv1" />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
