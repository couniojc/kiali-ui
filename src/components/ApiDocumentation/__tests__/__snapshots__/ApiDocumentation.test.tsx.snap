// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiDocumentation renders when not known 1`] = `
<div>
  <rapi-doc
    allow-authentication="false"
    allow-try="false"
    layout="column"
    show-header="false"
    show-info="false"
    spec-url="api/namespaces/ns1/services/serv1/apispec"
  />
</div>
`;
